src/sentry/grouping/enhancer/matchers.py

```diff
@@ -59,34 +61,60 @@ def _get_function_name(frame_data: dict, platform: Optional[str]):
 
 def create_match_frame(frame_data: dict, platform: Optional[str]) -> dict:
     """ Create flat dict of values relevant to matchers """
-    match_frame = dict(
-        category=get_path(frame_data, "data", "category"),
-        family=get_behavior_family_for_platform(frame_data.get("platform") or platform),
-        function=_get_function_name(frame_data, platform),
-        in_app=frame_data.get("in_app"),
-        module=get_path(frame_data, "module"),
-        package=frame_data.get("package"),
-        path=frame_data.get("abs_path") or frame_data.get("filename"),
-    )
+    return frame_data
 
-    for key in list(match_frame.keys()):
-        value = match_frame[key]
-        if isinstance(value, (bytes, str)):
-            if key in ("package", "path"):
-                value = match_frame[key] = value.lower()
 
-            if isinstance(value, str):
-                match_frame[key] = value.encode("utf-8")
+@dataclass
+class MatchConfig:
+    field: str
+    path_like: bool
+    pattern_is_glob: bool
+    negated: bool
+    patterns: Sequence[str]
+    required: bool = False
 
-    return match_frame
 
-
-class Match:
+class Match(ABC):
     description = None
 
+    @abstractmethod
+    def extract_frame_data(self, frames, idx, platform, exception_data):
```

Comment:
Extracting frame data per matcher means potentially executing expensive functions (e.g. `get_function_name_for_frame`) more than once per frame. Will this be compensated by the move to rust?